lmnet/lmnet/datasets/tfds.py

```diff
@@ -39,32 +46,40 @@ def __init__(
             **kwargs,
         )
 
-        self.session = tf.Session()
-
-        # The last directory name of extend_dir is the dataset name
-        splitted_path = self.extend_dir.rstrip('/').split('/')
-        data_dir = '/'.join(splitted_path[:-1])
-        name = splitted_path[-1]
-        builder = tfds.builder(name, data_dir=data_dir)
+        builder = tfds.builder(tfds_name, data_dir=tfds_data_dir)
+        if tfds_download:
+            builder.download_and_prepare()
 
         self.info = builder.info
         self._init_available_splits()
         self._validate_feature_structure()
 
-        self.dataset = builder.as_dataset(split=self.available_splits[self.subset])
-        self._init_features()
-
-    def __getitem__(self, i, type=None):
-        return self.features[i]
-
-    def __len__(self):
-        return self.num_per_epoch
+        self.tf_dataset = builder.as_dataset(split=self.available_splits[self.subset])
+        self.tf_dataset = self.tf_dataset.map(
+            lambda record: _label_to_one_hot(record, self.num_classes),
+            num_parallel_calls=tf.data.experimental.AUTOTUNE
+        )
 
     @property
     def num_per_epoch(self):
         split = self.available_splits[self.subset]
         return self.info.splits[split].num_examples
 
+    @property
+    def classes(self):
+        return self.info.features["label"].names
```

Comment:
Need to change when implement object detection dataset class, because `voc2007` builder use `labels` (not `label`) as label.
 https://github.com/tensorflow/datasets/blob/7dd97e23bd7cb5c07b5da19730b5cf377873af12/tensorflow_datasets/image/voc.py#L138
https://github.com/tensorflow/datasets/blob/master/docs/datasets.md#features-32

Will we change this line when we implement Object detection dataset class ?